src/NuGet.Core/NuGet.ProjectModel/PackageSpecExtensions.cs

```diff
@@ -42,5 +43,29 @@ public static ProjectRestoreMetadataFrameworkInfo GetRestoreMetadataFramework(th
 
             return frameworkInfo ?? new ProjectRestoreMetadataFrameworkInfo();
         }
+
+        /// <summary>
+        /// Get the <see cref="TargetFrameworkInformation"/> for the given target alias.
+        /// If there is not a match, null is returned.
+        /// </summary>
+        /// <param name="project">The project</param>
+        /// <param name="targetAlias">The target alias to search for</param>
+        /// <returns>The <see cref="TargetFrameworkInformation"/> if it exists, <see langword="null"/> otherwise. </returns>
+        public static TargetFrameworkInformation GetTargetFramework(this PackageSpec project, string targetAlias)
+        {
+            return project.TargetFrameworks.FirstOrDefault(f => StringComparer.Ordinal.Equals(targetAlias, f.TargetAlias));
```

Comment:
perf nitpick: the two methods added to this file both allocate enumerators and since the delegate captures a variable, there's 100% guaranteed delegate allocations as well.

Each of these allocations are tiny, but restore is our hot path, and all these tiny allocations can easily add additional GCs.